graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -398,46 +428,21 @@ def get_selectivity_of_filters_at_vertex(schema_info, filter_infos, parameters,
-            if filter_info.op_name == "in_collection":
-                if not filter_uses_only_runtime_parameters(filter_info):
-                    continue  # We can't reason about tagged parameters in in_collection filters
-
-                # TODO(bojanserafimov): Check if the filter values are in the interval selected
-                #                       by the inequality filters.
-                collection = parameters[get_parameter_name(filter_info.args[0])]
-                selectivity_per_entry_in_collection = _estimate_filter_selectivity_of_equality(
-                    schema_info, location_name, filter_info.fields
-                )
-
-                # Assumption: the selectivity is proportional to the number of entries in
-                # the collection. This will not hold in case of duplicates.
-                if _is_absolute(selectivity_per_entry_in_collection):
-                    selectivity = Selectivity(
-                        kind=ABSOLUTE_SELECTIVITY,
-                        value=float(len(collection)) * selectivity_per_entry_in_collection.value,
-                    )
-                elif _is_fractional(selectivity_per_entry_in_collection):
-                    selectivity = Selectivity(
-                        kind=FRACTIONAL_SELECTIVITY,
-                        value=min(
-                            float(len(collection)) * selectivity_per_entry_in_collection.value, 1.0
-                        )
-                        # The estimate may be above 1.0 in case of duplicates in the collection
-                        # so we make sure the value is <= 1.0
-                    )
-                selectivity_at_field = _combine_filter_selectivities(
-                    [selectivity_at_field, selectivity]
-                )
-
-        # Process equality filters
+        # Process in_collection and = filters
         for filter_info in filters_on_field:
-            if filter_info.op_name == "=":
-                # TODO(bojanserafimov): Check if the filter value is in the interval selected
-                #                       by the inequality filters.
-                selectivity = _estimate_filter_selectivity_of_equality(
-                    schema_info, location_name, filter_info.fields
+            if filter_info.op_name in ("=", "in_collection"):
+                filter_argument = get_only_element_from_collection(filter_info.args)
+                filter_field = get_only_element_from_collection(filter_info.fields)
+                collection = None
+                if is_runtime_parameter(filter_argument):
+                    # TODO(bojanserafimov): Check if the filter values are in the interval selected
+                    #                       by the inequality filters.
+                    collection = parameters[get_parameter_name(filter_argument)]
+                    if filter_info.op_name == "=":
+                        collection = [collection]
```

Comment:
`mypy` is going to have things to say about the type of `collection` changing. I think this module's `mypy` rules are pretty lax right now, but it might be easier to fix it now rather than fixing it later when we try to dial up the rule strictness. Your call.

Reply:
I'll come back when rules are dialed up. This section will require some casts/asserts too, so it's hard to blindly guess what passes.